src/System.IO.FileSystem.Watcher/src/System/IO/FileSystemWatcher.Linux.cs

```diff
@@ -470,7 +477,7 @@ private void ProcessEvents()
                         }
 
                         uint mask = nextEvent.mask;
-                        bool addWatch = false;
+                        bool addWatch = ((mask & (uint)AddMaskFilters) != 0);
```

Comment:
Can we move this down now to where it's used?  I'm not sure why I had it so far away.